x/auth/ante/unorderedtx/snapshotter.go

```diff
@@ -80,10 +80,11 @@ func (s *Snapshotter) restore(height uint64, payloadReader snapshot.ExtensionPay
 		copy(txHash[:], payload[i:i+txHashSize])
 
 		timestamp := binary.BigEndian.Uint64(payload[i+txHashSize : i+chunkSize])
-
-		// add all txs, we don't care at this point if they are expired,
-		// we'll let the purge loop handle that
-		if timestamp != 0 {
+		// need to come up with a way to fetch blocktime to filter out expired txs
+		//
+		// right now we dont have access block time at this flow, so we would just include the expired txs
+		// and let it be purge during purge loop
+		if timestamp != 0 && timestamp > height {
```

Comment:
Do you understand this? 

Looks like comparing a timestamp that is extracted from a block to a block height?

Reply:
looks like that was edited in the last audit change. i updated it to that. [update snapshotter code from latest audit](https://github.com/cosmos/cosmos-sdk/pull/23734/commits/14794e4e89624c9da5da87572b2f877a51d7a2a5)

Reply:
should make more sense now.